src/viewer.js

```diff
@@ -1199,30 +1869,40 @@ class VolumeImageViewer {
    * @param {string} uid - Unique identifier of the region of interest
    */
   removeROI(uid) {
-    console.info(`remove ROI ${uid}`)
+    console.info(`remove ROI ${uid}`);
     const feature = this[_drawingSource].getFeatureById(uid);
-    this[_features].remove(feature);
+
+    if (feature) {
+      this[_features].remove(feature);
+      return;
+    }
+
+    /** 
+     * If failed to draw/cache feature in drawing source, call onFailure
+     * to avoid trash of broken annotations
+     */
+    this[_annotationManager].onFailure(uid);
   }
 
   /** Removes all annotated regions of interest. */
   removeAllROIs() {
-    console.info('remove all ROIs')
+    console.info("remove all ROIs");
     this[_features].clear();
   }
 
   /** Hides annotated regions of interest such that they are no longer
    *  visible on the viewport.
    */
   hideROIs() {
-    console.info('hide ROIs')
+    console.info("hide ROIs");
     this[_drawingLayer].setVisible(false);
```

Comment:
When this method is called the measurements associated with the ROI remain visible.

![screenshot_slim_ROI_show](https://user-images.githubusercontent.com/6855297/113781074-2023aa80-96fe-11eb-93e5-313f602a008a.png)

![screenshot_slim_ROI_hide_bug](https://user-images.githubusercontent.com/6855297/113781067-1c902380-96fe-11eb-8d14-e8f2afe003f6.png)

Reply:
This issue is still persisting. I think we will either need to use the same VectorLayer (`_drawingLayer` stored on viewer object) for ROIs and markups or expose the vector layer(s) created by the markup manager to the viewer so that the `hideROIs()` and `showROIs()` methods can set the visibility of all the vector layers.

Reply:
Probably better to keep them on separate vector layers. That will facilitate toggling markers and markups separately.